### ISSUE TITLE
Root Element Missing, on ExecuteQuery

### ISSUE THREAD

**Issue by TheWanderingAsian** [SharePoint/sp-dev-docs#9336]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

SharePoint CSOM

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [X] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version
- SPFx version
- Node.js version
- etc

### Describe the bug / error

When interacting with the SharePoint, calling ExecuteQuery to retreive ANY information errors out with "Root Element Missing".

The code in question was running perfectly fine for almost a year until week ago friday, when it started to error.

### Steps to reproduce

1. Write any code to load the context for e.g web.Context.Load(web, w => w.ServerRelativeUrl);
2. Execute the query to materialise the results web.Context.ExecuteQuery();
3. error

### Expected behavior

To return the properties or data from the loaded context

**Comment by craig-blowfield** [comment]
We are also having this issue.

Our solution is hosted within an Azure Function using User credentials (MFA is not enabled) to provision elements in SharePoint.

FYI here is the relevant part of the stack trace

   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.XmlTextReaderImpl.Read()
   at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
   at Microsoft.SharePoint.Client.Idcrl.IdcrlAuth.DoGet(String url)
   at Microsoft.SharePoint.Client.Idcrl.IdcrlAuth.RequestFederationProviderInfo(String domainname)
   at Microsoft.SharePoint.Client.Idcrl.IdcrlAuth.GetFederationProviderInfo(String domainname)
   at Microsoft.SharePoint.Client.Idcrl.IdcrlAuth.InitFederationProviderInfoForUser(String username)
   at Microsoft.SharePoint.Client.Idcrl.IdcrlAuth.GetServiceToken(String username, String password, String serviceTarget, String servicePolicy)
   at Microsoft.SharePoint.Client.Idcrl.SharePointOnlineAuthenticationProvider.GetAuthenticationCookie(Uri url, String username, SecureString password, Boolean alwaysThrowOnFailure, EventHandler`1 executingWebRequest)
   at Microsoft.SharePoint.Client.SharePointOnlineCredentials.GetAuthenticationCookie(Uri url, Boolean refresh, Boolean alwaysThrowOnFailure)
   at Microsoft.SharePoint.Client.ClientRuntimeContext.SetupRequestCredential(ClientRuntimeContext context, HttpWebRequest request)
   at Microsoft.SharePoint.Client.SPWebRequestExecutor.GetRequestStream()
   at Microsoft.SharePoint.Client.ClientContext.GetWebRequestExecutor()
   at Microsoft.SharePoint.Client.ClientContext.<GetFormDigestInfoPrivateAsync>d__37.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.SharePoint.Client.ClientContext.<EnsureFormDigestAsync>d__36.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.SharePoint.Client.ClientContext.<ExecuteQueryAsync>d__28.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__7.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryRetry(ClientRuntimeContext clientContext, Int32 retryCount, Int32 delay, String userAgent)
   at

**Comment by TheWanderingAsian** [comment]
Hi Craig, I've just solved this issue last night, we had a similar auth setup - the solution for me was this method of authentication seems to be depreciated, with now certificate based auth being required.

So I followed the steps here:
https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azuread

Generating the cert, uploading into azure into a new app registration and using pfx to auth in code.

I suppose if you needed the code to impersonate a user without intervention this wouldn't work, but if you just need a background service to interact you should be fine.

Good luck!

**Comment by craig-blowfield** [comment]
> Hi Craig, I've just solved this issue last night, we had a similar auth setup - the solution for me was this method of authentication seems to be depreciated, with now certificate based auth being required.
> 
> So I followed the steps here: https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azuread
> 
> Generating the cert, uploading into azure into a new app registration and using pfx to auth in code.
> 
> I suppose if you needed the code to impersonate a user without intervention this wouldn't work, but if you just need a background service to interact you should be fine.
> 
> Good luck!

Thanks @TheWanderingAsian for your help. Appreciated

Is there anywhere that states this method of auth has been deprecated.

Looking through a network trace during authentication I can see the following request failing

GET http://msoid.mydomain.co.uk/FPUrl.xml HTTP/1.1
Host: msoid.proaktive.co.uk
Proxy-Connection: keep-alive
DNT: 1
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9

With response 

HTTP/1.1 502 Fiddler - DNS Lookup Failed
Date: Fri, 24 Nov 2023 10:14:01 GMT
Content-Type: text/html; charset=UTF-8
Connection: close
Cache-Control: no-cache, must-revalidate
Timestamp: 10:14:01.703

[Fiddler] DNS Lookup for "msoid.mydomain.co.uk" failed. System.Net.Sockets.SocketException No such host is known                                                                                                                                                                                                                                                                                                                                                                                                               

So people elesewhere have suggested updating DNS settings

http://paulryan.com.au/2014/for-security-reasons-dtd-is-prohibited-in-this-xml-document/

Also at the time (many years ago) if I remember correctly the reason why I 'had' to user user credentials to auth was due to either Creating a Modern Team or Communication site didn't allow the use of app only credentials (https://github.com/pnp/PnP-Sites-Core/issues/1481)

**Comment by TheWanderingAsian** [comment]
No worries, here's a message from MS about it:

![image](https://github.com/SharePoint/sp-dev-docs/assets/47541753/53a26431-11b3-467d-ba26-b7e114422596)

From the bottom of the link I sent in the FAQ, it's not really a concrete message of depreciation though.

Yeah I saw the DNS answers too, I tried those and in my case it didn't help though it's worth giving it a go in yours if you have access to a staging env?

I don't have any experience with communication sites unfortunately, only the bog standard ones - though I reckon it could be supported now as the issue you quoted is from a while back. I also saw some reference to provisioning communication sites using the Graph Rest API:
https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/modern-experience-customizations-provisioning-sites#provision-a-modern-team-site-programmatically-via-sharepoint-online-rest-api

Also the PnP sites Core package is depreciated in favour of PnP Framework / PnP Core, so that may offer some additional support, if you were to swap it out.

https://github.com/pnp/pnpframework
